enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,573 @@
+
+### Goals
+
+* Enable an in-cluster workflow for snapshotting and restoration of cluster workload data
+* Follow the [MicroShift design principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* Enable backup/restore workflows external to MicroShift
+* Provide a workflow for exporting data from a MicroShift cluster
+* Enabling MicroShift deployment without CSI Snapshotter.  This falls under platform composability and is out of scope
+* Describe MicroShift installation processes
+
+## Proposal
+
+Deploy the CSI snapshot controller and CSI plugin sidecar with MicroShift out of the box to provide users with a means of 
+snapshotting, cloning, and restoring workload state.
+
+### Workflow Description
+
+#### Deploy MicroShift with Snapshotting Support
+
+_Prerequisites_
+
+* The device owner has created an LVM volume group and a thin-pool on this volume group.
+* The `/etc/microshift/lvmd.yaml` config includes a `deviceClass` to represent the thin-pool.  The lvmd.yaml may 
+contain a mix of thin and thick `deviceClasses`.
+
+```yaml
+device-classes:
+- name: ssd-thin
+  volume-group: myvg1
+  type: thin
+  thin-pool:
+    name: pool0
+    overprovision-ratio: 50.0
+- name: default
+  volume-group: rhel
+  default: true
+  spare-gb: 0
+```
+
+_Workflow_
+
+1. The user installs the MicroShift RPMs and starts the MicroShift systemd service.
+2. The Service Manager initiates the `startCSIPlugin()` service.
+3. The `startCSIPlugin()` service deploys the CSI VolumeSnapshot CSI Snapshot Controller and Validating Webhook, and a 
+   default VolumeSnapshotClass.
+   - The `startCSIPlugin()` starts LVMS immediately after. It is not necessary to wait for the CSI Controller to 
+         become **Ready** as there are no start-sequence dependencies between the 2 systems.
+4. The CSI VolumeSnapshot Controller begins listening for VolumeSnapshot API events.
+5. The user observes the CSI Snapshot Controller pod reaches the **Ready** state.
+   - `$ oc get pod -n kube-system csi-snapshot-controller-$SUFFIX`
+6. The user observes the `topolvm-controller` and `csi-snapshot-controller` pods reach the **Ready** state.
+    - `$ oc get pod -n openshift-storage topolvm-controller-$SUFFIX`
+    - The `topolvm-controller` includes the `csi-external-snapshotter` sidecar container, which is why the user must 
+      verify its phase.
+      
+#### Create a Snapshot Dynamically from a PVC
+
+This section describes how to create snapshot from a PVC utilizing a VolumeSnapshotClass to dynamically specify 
+provisioning parameters.
+
+_Prerequisites_
+
+* A running MicroShift cluster
+* CSI Snapshot Controller is in **Ready** state
+* Topolvm-Controller is in **Ready** state
+* `volumeSnapshotClass` has been deployed
+* `storageClass` has been deployed
+
+_Workflow_
+ 
+1. The user creates a PVC of an arbitrary size, in Gb increments, to be the source volume from which we will 
```

Comment:
Need to make it clear that 1-5 are preliminary steps a user would follow to deploy a workload.